### ISSUE TITLE
Edit Protocol feature does not work with Java > 14

### ISSUE THREAD

**Issue by ferpasri** [TESTARtool/TESTAR_dev#341]
Java Nashorn Script Engine was included in the Java JDK from Java 8 to Java 14.  
This feature allows the execution of the Edit Protocol.  

However, it was removed from JDK 15 onwards.  For this reason, if the user tries to use the GUI "Edit Protocol" feature, TESTAR throws an exception `Cannot invoke "javax.script.ScriptEngine"`

![image](https://user-images.githubusercontent.com/32359126/224939863-a7a9864a-5f83-4612-8238-532ecd754ed1.png)

**TEMP SOLUTION:**
It is still possible to Edit the file protocols accessing the java files in: `testar\bin\settings\`  
TESTAR will compile this file and execute the changes.

**Comment by adibkhaled** [comment]
If you comment out the code below, it will fix the issue for Java 15. The only problem is that your edited protocol text will not be highlighted any more.

<img width="728" alt="Screenshot 2023-04-15 at 17 14 48" src="https://user-images.githubusercontent.com/3850762/232233672-0d2313da-307c-4abe-814c-cd08eda33398.png">

**Comment by ferpasri** [comment]
Thanks.
We can maybe check if Nashorn Script Engine is detected in the JVM. Then enable or not the `DefaultSyntaxKit.initKit()` feature.

```
public ProtocolEditor(String settingsDir, String protocolClass) {
    this.settingsDir = settingsDir;
    this.protocolClass = protocolClass;
    enableDefaultSyntaxKit();
    initComponents();
    codeEditor.setContentType("text/java");
    codeEditor.setText(Util.readFile(getProtocolFile()));
}

/**
 * Enable DefaultSyntaxKit when Nashorn Script Engine is detected in the JVM. 
 * If not, inform the user in the console and start Edit Protocol without code highlights. 
 */
private void enableDefaultSyntaxKit() {
    try {
        ScriptEngineManager manager = new ScriptEngineManager();
    	ScriptEngine engine = manager.getEngineByName("nashorn");
    	if (engine != null) {
    	    DefaultSyntaxKit.initKit();
        } else {
    	    System.out.println("WARNING: Edit Protocol does not highlight the Java code because Nashorn Script Engine is not detected");
    	}
    } catch (Exception e) {
    	System.err.println("ERROR: Trying to verify Nashorn Script Engine with ScriptEngineManager");
    }
}
```

**Comment by ferpasri** [comment]
Fixed with PR #356 in TESTAR v2.6.4
